keras/export/export_lib_test.py

```diff
@@ -687,18 +687,25 @@ def test_reloading_export_archive(self):
         export_lib.export_model(model, temp_filepath)
         reloaded_layer = export_lib.TFSMLayer(temp_filepath)
         self.assertAllClose(reloaded_layer(ref_input), ref_output, atol=1e-7)
-        self.assertLen(reloaded_layer.weights, len(model.weights))
         self.assertLen(
-            reloaded_layer.trainable_weights, len(model.trainable_weights)
+            reloaded_layer._reloaded_obj.variables, len(model.variables)
```

Comment:
What's the reason the reloaded collections are not exposed at the layer level? That will break the fine-tuning use case. If it doesn't work with JAX, we should error out explicitly upon access to an unavailable collection, and we should keep testing the correct behavior with TF.